src/Bicep.Local.Deploy/LocalDeploymentEngine.cs

```diff
@@ -0,0 +1,189 @@
+    {
+        using (RequestCorrelationContext.NewCorrelationContextScope())
+        {
+            RequestCorrelationContext.Current.Initialize(apiVersion: requestContext.ApiVersion);
+
+            var (template, parameters) = ParseTemplateAndParameters(templateString, parametersString);
+
+            if (template.Resources.Any(x => x.Import is null))
+            {
+                throw new NotImplementedException("Only resources with imports are supported");
+            }
+
+            var context = DeploymentContextWithScopeDefinition.CreateAtResourceGroup(
+                tenantId: Guid.Empty.ToString(),
+                subscriptionId: Guid.Empty.ToString(),
+                resourceGroupName: "mockRg",
+                resourceGroupLocation: requestContext.Location,
+                deploymentName: "main",
+                subscriptionDefinition: null,
+                resourceGroupDefinition: null,
+                tenantDefinition: null);
+
+            var definition = new DeploymentContent
+            {
+                Properties = new DeploymentPropertiesContent
+                {
+                    Template = template,
+                    Parameters = parameters,
+                    Mode = DeploymentMode.Incremental,
+                },
+            };
+
+            var oboToken = "dummyToken";
+            var oboCorrelationId = RequestCorrelationContext.Current.CorrelationId;
+
+            var deploymentPlan = await this.deploymentEngine.ProcessDeployment(
+                preflightSettings: new(settings, syncMode: true),
+                deploymentContext: context,
+                definition: definition,
+                originalDeploymentName: context.DeploymentName,
+                isNestedDeployment: false,
+                validateOnly: false,
+                request: null,
+                frontdoorEndpointUri: new Uri(requestContext.FrontdoorEndpoint),
+                cancellationToken: cancellationToken,
+                oboToken: oboToken,
+                oboCorrelationId: oboCorrelationId);
+
+            await StartDeployment(deploymentPlan);
+
+            var dataProvider = dataProviderHolder.GetDeploymentDataProvider(deploymentPlan.DeploymentLocation);
+            return await WaitForDeploymentToComplete(dataProvider, context, cancellationToken);
+        }
+    }
+
+    private async Task<LocalDeployment.Result> WaitForDeploymentToComplete(IDeploymentDataProvider dataProvider, DeploymentContext context, CancellationToken cancellationToken)
+    {
+        var entity = await dataProvider.FindDeployment(context.SubscriptionId, context.ResourceGroupName, context.DeploymentName);
+        while (!entity.ProvisioningState.IsTerminal())
+        {
+            await Task.Delay(500, cancellationToken);
+
+            entity = await dataProvider.FindDeployment(context.SubscriptionId, context.ResourceGroupName, context.DeploymentName);
```

Comment:
It would be nice to avoid polling here - it's a consequence of low visibility into ResourceStack's job processing internals. I've cut the polling down to every 20ms to make it more responsive as it should be very low-cost.